test/test_s3_utils.py

```diff
@@ -529,13 +579,24 @@ def test_read_s3_zip_integrated(integrated_s3_info):
 @pytest.mark.unit
 def test_read_s3_zip_unit(integrated_names):
 
-    with mocked_s3_integration(integrated_names=integrated_names) as s3_connection:
+    # with mocked_s3_integration(integrated_names=integrated_names) as s3_connection:
+    ffenv = integrated_names['ffenv'] if integrated_names else None
+
+    with mocked_s3utils_with_sse(beanstalks=[ffenv], require_sse=False):
+        s3_connection = s3Utils(env=ffenv)
+
+        # In our mock, this won't exist already on S3 like in the integrated version of this test,
+        # so we have to pre-load to our mock S3 manually. -kmp 13-Jan-2021
+        s3_connection.s3.upload_file(Filename=integrated_names['zip_path'],
+                                     Bucket=s3_connection.outfile_bucket,
+                                     Key=integrated_names['zip_filename'])
 
```

Comment:
As stated previously, `s3Utils` does not implement a true interface to `s3`. This code while mocked appropriately would not function in production as it does not pass KMS arguments. Similar issue in other places here.